fetcher/catchupservice.go

```diff
@@ -0,0 +1,269 @@
+package fetcher
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/algorand/go-algorand/agreement"
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/bookkeeping"
+	"github.com/algorand/go-algorand/logging"
+	"github.com/algorand/go-algorand/network"
+	"github.com/algorand/go-algorand/rpcs"
+	"github.com/algorand/indexer/util/metrics"
+)
+
+// CatchupService represents the catchup service implemented to enable indexer to directly fetch blocks from the network
+type CatchupService struct {
+	net          network.GossipNode
+	log          logging.Logger
+	cfg          config.Local
+	genesis      bookkeeping.Genesis
+	peerSelector *peerSelector
+}
+
+// nodeInfo has context and implements one service required to create a gossip node
+type nodeInfo struct {
+	ctx context.Context
+}
+
+type task func() basics.Round
+
+// makeNodeInfo initializes nodeInfo.
+func makeNodeInfo(ctx context.Context) nodeInfo {
+	return nodeInfo{
+		ctx: ctx,
+	}
+}
+
+func (n nodeInfo) IsParticipating() bool { return false }
+
+// MakeCatchupService creats a catchup service and initialzes a gossipnode
+func MakeCatchupService(ctx context.Context, genesis bookkeeping.Genesis) (s *CatchupService) {
+	s = &CatchupService{}
+
+	s.cfg = config.AutogenLocal
+	s.genesis = genesis
+	s.log = logging.NewLogger()
+	nodeinfo := makeNodeInfo(ctx)
+	s.net, _ = network.NewWebsocketNetwork(s.log, s.cfg, nil, genesis.ID(), genesis.Network, nodeinfo)
```

Comment:
making a websocket network and connecting to relays means by default, you are going to ask all those relays to send you transactions, votes, and proposals it sees via the websocket gossip connection. But since all you care about is the /v1/{network}/block/N HTTP endpoint, you can get away with not doing any of the websocket stuff... 

To filter out gossip messages there is a "MessagesOfInterest" map that you can set to be empty so that the relays don't try to send you any gossip messages.

Reply:
`MessagesOfInterest` map is set during `setup()` through `RegisterMessageInterest` if `relayMessages` boolean is set to true (nominally true for relays, false otherwise). `config.AutogenLocal` leads to `relayMessages` being false, so the default value of map is already empty.

Reply:
the default set that peers will send you is actually `defaultSendMessageTags` unless you deregister your interest. Otherwise you are going to be sent gossip messages (like votes, transactions, etc) that you'll drop anyway (because you don't have any services calling wn.RegisterHandlers).